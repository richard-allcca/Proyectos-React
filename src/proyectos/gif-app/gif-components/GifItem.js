import React from "react";
import Prototypes from "prop-types";
// animation
import "animate.css";

const GifItem = ({ title, images }) => {
  const { url } = images.downsized_medium;
  return (
    <div className="card animate__animated animate__bounce animate__fadeIn">
      <img src={url} alt={title} />
      <h3>{title}</h3>
    </div>
  );
};

export default GifItem;

GifItem.prototypes = {
  title: Prototypes.string.isRequired,
};

// Notas:
// Props: traidas con spread operators desde (ListCategories)
