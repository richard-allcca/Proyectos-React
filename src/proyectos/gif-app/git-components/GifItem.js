import React from "react";
// animation
import "animate.css";

const GifItem = ({ title, id, images }) => {
  const { downsized_medium } = images;
  const { url } = downsized_medium;
  return (
    <div className="card animate__animated animate__bounce animate__fadeIn">
      <img src={url} alt={title} />
      <h3>{title}</h3>
    </div>
  );
};

export default GifItem;

// Notas:
// Props: traidas con spread operators desde (ListCategories)
