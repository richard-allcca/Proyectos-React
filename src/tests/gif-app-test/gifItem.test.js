import { shallow } from "enzyme";
import React from "react";
import GifItem from "../../proyectos/gif-app/gif-components/GifItem";

describe("Prueba del componente GifItem", () => {
  let title = "titulo";
  let wrapper = shallow(<GifItem title={title} />);

  // ? esto es para reiniciar el wrapper en caso le hagas cambios al component original
  beforeEach(() => {
    wrapper = shallow(<GifItem title={title} />);
  });

  test("Retorna contenido del componente GifItem", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe retornar un title", () => {
    const h3 = wrapper.find("h3");
    expect(h3.text().trim()).toBe(title);
  });

  test("debe de tener animate__fadeIn", () => {
    const div = wrapper.find("div");
    const className = div.prop("className");

    expect(className.includes("animate__fadeIn")).toBe(true);
    //  expect(className).toContain("animate__fadeIn");
  });
});

// Notas:
//! el test al componente GifItem no se completo, aun no se resuelve el 'images.downsized_medium'
// en este test no podemos validar el contenido de 'images'
// falta mas detalles de como hacer este test para contenido de props dinamicos
