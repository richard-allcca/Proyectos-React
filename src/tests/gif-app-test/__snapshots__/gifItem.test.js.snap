// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del componente GifItem Retorna contenido del componente GifItem 1`] = `
<div
  className="card animate__animated animate__bounce animate__fadeIn"
>
  <img
    alt="titulo"
    src="#"
  />
  <h3>
    titulo
  </h3>
</div>
`;
